app/validators/empty_profile_field_names_validator.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class EmptyProfileFieldNamesValidator < ActiveModel::Validator
+  def validate(account)
+    return if account.fields.empty?
+
+    field_names_valid = true
+    account.fields.each do |field|
+      field_names_valid = false if field.name.blank? && field.value.present?
+    end
+    return if field_names_valid
```

Comment:
Instead of the local var and explicit return here, might be a little cleaner to have something like `unless fields_with_values_missing_names?(account) ...` (or whatever on naming) and relying on one path of that to add the errors and an implicit return to not add when all is well?

Reply:
Thanks! Still getting used to Ruby semantics. I've updated this accordingly, please let me know if that doesn't match your intention.